README.md

```diff
@@ -50,3 +50,57 @@ Test your new function; I've provided some basic testing code in `code.test.js`.
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
 reasoning, to this markdown file.
+
+Recall my code,
+```js
+function tsp_ls(distance_matrix) {
+  let n = distance_matrix.length;
+  let minRoute = Array.from({ length: n }, (v, i) => i).sort((a, b) => 0.5 - Math.random());// This is memory complexity of O(n)
+  let minCost = calcCost(minRoute, distance_matrix); // This is time complexity of O(n) and memory complexity of O(1)
+  let improving = true;
+  // In the worst case, this is time complexity of \Theta(n^2) because it'll loop through every possible r and l choice in the order of worst to best 
+  while(improving) { 
+    let routePath = [...minRoute]; // Get a deep copy of min route
+    let r = Math.floor(Math.random() * (n - 1)) + 1; // Random Number in the range 1 to n - 1
+    let l = Math.floor(Math.random() * (n - r - 1)) + (r + 1); // Random number in the range r + 1 to n
+    swap(routePath, r, l);// This is time complexity of \Theta(n)
+    let newCost = calcCost(routePath, distance_matrix); // This is O(n) time complexity
+    if (newCost < minCost) {
+      minCost = newCost;
+      minRoute = routePath;
+      improving = true;
+    }
+    else improving = false; // no improvement
+  }
+  return minCost;
+}
+ function calcCost(route, distance_matrix){ // This is time complexity of \Theta(n)
+    let newCost = 0;
+    for (let i = 0; i < distance_matrix.length - 1; i++) {
+      newCost += distance_matrix[route[i]][route[i + 1]];
+    }
+    return newCost;
+ }
+function swap(arr, r, l) { // This is time complexity of \Theta(n)
+  while (r < l) {
+    let temp = arr[r];
+    arr[r] = arr[l];
+    arr[l] = temp;
+    r++;
+    l--;
+  }
+}
+```
+Adding those up
+- Time complexity of $\Theta((n + n^2 * (n + n))) \in \Theta(n^3)$
```

Comment:
Not sure how you're getting this complexity -- what's your termination condition?

Reply:
I just put this in my readme but the while loop terminates when no improvement is found in a single iteration. In the worst case, it may explore all possible swaps of two cities,  which is ${{n}\choose{2}} = \frac{n(n-1)}{2} \in \Theta(n^2)$. Also note that each iteration involves calculating the cost of the route O(n)
and performing a swap O(n).

Reply:
But the same swap may result in a different route, depending on the route before the swap, right? Wouldn't that affect the overall complexity?